tasks/forms.py

```diff
@@ -139,6 +129,24 @@ def init_layout(self):
             ),
         )
 
+    def update_status(self, task) -> bool:
```

Comment:
Perhaps name this function to reflect which field it's updating?
```suggestion
    def update_progress_state(self, task: Task) -> bool:
```